### ISSUE TITLE
`pipenv install django-environ` fails at `do_lock()` with `KeyError` and loses lock file.

### ISSUE THREAD

**Issue by tveastman** [pypa/pipenv#2072]
Running pipenv from `brew install pipenv` on OS X. I'm trying to add a package to an uncomplicated Django project, I'll attach the pipfile.

```
# Note that I'm running that brew's "wrapper" script executes here.
$ PATH="/usr/local/Cellar/pipenv/11.10.0/libexec/tools:$PATH" "/usr/local/Cellar/pipenv/11.10.0/libexec/bin/python" -m pipenv.help
<details><summary>$ python -m pipenv.help output</summary>

Pipenv version: `'11.10.0'`

Pipenv location: `'/usr/local/Cellar/pipenv/11.10.0/libexec/lib/python3.6/site-packages/pipenv'`

Python location: `'/usr/local/Cellar/pipenv/11.10.0/libexec/bin/python'`

Other Python installations in `PATH`:

  - `2.7`: `/usr/local/bin/python2.7`
  - `2.7`: `/usr/local/bin/python2.7`
  - `2.7`: `/usr/local/bin/python2.7`
  - `2.7`: `/usr/local/bin/python2.7`
  - `2.7`: `/usr/bin/python2.7`
  - `3.6`: `/usr/local/bin/python3.6m`
  - `3.6`: `/usr/local/bin/python3.6`
  - `3.6`: `/usr/local/bin/python3.6`
  - `3.6`: `/usr/local/bin/python3.6`

  - `2.7.14`: `/usr/local/bin/python`
  - `2.7.14`: `/usr/local/bin/python`
  - `2.7.14`: `/usr/local/bin/python`
  - `2.7.10`: `/usr/bin/python`
  - `2.7.14`: `/usr/local/bin/python2`
  - `2.7.14`: `/usr/local/bin/python2`
  - `2.7.14`: `/usr/local/bin/python2`
  - `3.6.5`: `/usr/local/bin/python3`
  - `3.6.5`: `/usr/local/bin/python3`
  - `3.6.5`: `/usr/local/bin/python3`

PEP 508 Information:

```

```
{'implementation_name': 'cpython',
 'implementation_version': '3.6.5',
 'os_name': 'posix',
 'platform_machine': 'x86_64',
 'platform_python_implementation': 'CPython',
 'platform_release': '17.5.0',
 'platform_system': 'Darwin',
 'platform_version': 'Darwin Kernel Version 17.5.0: Mon Mar  5 22:24:32 PST '
                     '2018; root:xnu-4570.51.1~1/RELEASE_X86_64',
 'python_full_version': '3.6.5',
 'python_version': '3.6',
 'sys_platform': 'darwin'}
```

System environment variables:

  - `PATH`
  - `VIRTUALENVWRAPPER_WORKON_CD`
  - `VIRTUALENVWRAPPER_HOOK_DIR`
  - `LANG`
  - `XPC_FLAGS`
  - `DISPLAY`
  - `OLDPWD`
  - `TERM_SESSION_ID`
  - `WORKON_HOME`
  - `COLORTERM`
  - `PERL_MB_OPT`
  - `__CF_USER_TEXT_ENCODING`
  - `ITERM_SESSION_ID`
  - `SSH_AUTH_SOCK`
  - `USER`
  - `DRONE_TOKEN`
  - `PWD`
  - `HOME`
  - `ATLAS_TOKEN`
  - `TERM_PROGRAM`
  - `TERM_PROGRAM_VERSION`
  - `PGDATA`
  - `Apple_PubSub_Socket_Render`
  - `ITERM_PROFILE`
  - `PERL_MM_OPT`
  - `TMPDIR`
  - `PERL_LOCAL_LIB_ROOT`
  - `VIRTUALENVWRAPPER_PYTHON`
  - `XPC_SERVICE_NAME`
  - `VIRTUALENVWRAPPER_SCRIPT`
  - `TERM`
  - `SHELL`
  - `VAULT_AUTH_GITHUB_TOKEN`
  - `PERL5LIB`
  - `VAULT_ADDR`
  - `SHLVL`
  - `COLORFGBG`
  - `LOGNAME`
  - `DRONE_SERVER`
  - `VIRTUALENVWRAPPER_PROJECT_FILENAME`
  - `HOMEBREW_GITHUB_API_TOKEN`
  - `_`
  - `PYTHONDONTWRITEBYTECODE`
  - `PIP_PYTHON_PATH`
  - `PYTHONUNBUFFERED`

Pipenv–specific environment variables:

Debug–specific environment variables:

  - `PATH`: `/usr/local/Cellar/pipenv/11.10.0/libexec/tools:/Users/tom/.local/bin:/Users/tom/perl5/bin:/usr/local/opt/coreutils/libexec/gnubin:/usr/local/bin:/usr/local/opt/coreutils/libexec/gnubin:/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin:/opt/X11/bin`
  - `SHELL`: `/usr/local/bin/bash`
  - `LANG`: `en_NZ.UTF-8`
  - `PWD`: `/usr/local/Cellar/pipenv/11.10.0/libexec/bin`

---------------------------

</details>
```

Here's the Pipfile I'm starting from:

```[[source]]

url = "https://pypi.python.org/simple"
verify_ssl = true
name = "pypi"


[packages]

django = "*"
gunicorn = "*"
dj-database-url = "*"
"psycopg2-binary" = "*"
whitenoise = "*"
djangorestframework = "*"
coreapi = "*"
salt-pepper = "*"
hashids = "*"
hvac = "*"
django-fieldsignals = "*"


[dev-packages]

pytest = "*"
pytest-django = "*"
pytest-watch = "*"
pytest-sugar = "*"


[requires]

python_version = "3.6"


[pipenv]

keep_outdated = true

```

Here's the error:

```
$ pipenv install django-environ
Installing django-environ…
Looking in indexes: https://pypi.python.org/simple
Collecting django-environ
  Using cached https://files.pythonhosted.org/packages/0e/04/8a2b9d21ed73761b8e12201aa0531c0b0971fe8a832c9311cb3c2529fa98/django_environ-0.4.4-py2.py3-none-any.whl
Requirement already satisfied: django in /Users/tom/.virtualenvs/hakea-l_wmlrrm/lib/python3.6/site-packages (from django-environ) (2.0.4)
Requirement already satisfied: six in /Users/tom/.virtualenvs/hakea-l_wmlrrm/lib/python3.6/site-packages (from django-environ) (1.11.0)
Requirement already satisfied: pytz in /Users/tom/.virtualenvs/hakea-l_wmlrrm/lib/python3.6/site-packages (from django->django-environ) (2018.4)
Installing collected packages: django-environ
Successfully installed django-environ-0.4.4

Adding django-environ to Pipfile's [packages]…
Pipfile.lock (c74aff) out of date, updating to (5f8018)…
Locking [dev-packages] dependencies…
Locking [packages] dependencies…
Traceback (most recent call last):
  File "/usr/local/Cellar/pipenv/11.10.0/libexec/bin/pipenv", line 11, in <module>
    load_entry_point('pipenv==11.10.0', 'console_scripts', 'pipenv')()
  File "/usr/local/Cellar/pipenv/11.10.0/libexec/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/Cellar/pipenv/11.10.0/libexec/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/Cellar/pipenv/11.10.0/libexec/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/Cellar/pipenv/11.10.0/libexec/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/Cellar/pipenv/11.10.0/libexec/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/Cellar/pipenv/11.10.0/libexec/lib/python3.6/site-packages/pipenv/cli.py", line 366, in install
    selective_upgrade=selective_upgrade,
  File "/usr/local/Cellar/pipenv/11.10.0/libexec/lib/python3.6/site-packages/pipenv/core.py", line 1981, in do_install
    requirements_dir=requirements_directory,
  File "/usr/local/Cellar/pipenv/11.10.0/libexec/lib/python3.6/site-packages/pipenv/core.py", line 1321, in do_init
    do_lock(system=system, pre=pre, keep_outdated=keep_outdated)
  File "/usr/local/Cellar/pipenv/11.10.0/libexec/lib/python3.6/site-packages/pipenv/core.py", line 1142, in do_lock
    norm_name
KeyError: 'django-environ'
```

After this has run (and failed), my `Pipfile.lock` file has vanished.

I tried one or two other packages, and they seem to work, could there be something weird about `django-environ` or its dependencies causing the breakage?

**Comment by techalchemy** [comment]
Homebrew installed pipenv tends to fail pretty horribly, I highly recommend using a `pip install --user` installation of pipenv.  To be perfectly honest we don't usually even spend much time trying to debug these issues when homebrew is involved, it's just a nightmare.

**Comment by uranusjr** [comment]
Note: This does not mean we don’t want to support Homebrew (we do), but that no current maintainers use this, and we have no extra resources to support it. Feel free to help maintain it if you’d like to.

**Comment by tveastman** [comment]
I uninstalled the homebrew-installed version of `pipenv`, and installed the same version from pip: `pip3 install pipenv==11.10.0` and re-ran `pipenv install django-environ`.

I got the same error. 

However, after upgrading (`pip3 install --upgrade pipenv`) to `11.10.1` the installation appears to have been successful, so maybe it was a bug in `11.10.0` that has already been fixed.

> Note: This does not mean we don’t want to support Homebrew (we do), but that no current maintainers use this, and we have no extra resources to support it. Feel free to help maintain it if you’d like to.

I have no special need to use Homebrew, but I note that it is the top suggested installation method in the `pipenv` docs. If it's a nightmare, the devs don't use it, and don't support it, we should update the docs to reflect that `pip` is the preferred installation method. I'll open an issue to that effect.
